README.md
# hermes-bootstrap4


documentation/CHANGELOG.md
## [1.0.0] 2017-05-16
### Original Release

js/lbd.js

var navbar_initialized = false;

$(document).ready(function(){
	window_width = $(window).width();

	lbd.checkHeaderImage();
	
	if(window_width <= 991){
		lbd.initRightMenu();
	}
});

$(window).resize(function(){
	if($(window).width() <= 991){
		lbd.initRightMenu();
	}
})

lbd = {

	misc:{
		navbar_menu_visible: 0 
	},

	checkHeaderImage: function(){
		$pageheader = $('.page-header');
		image_src = $pageheader.data('image');

		if(image_src !== undefined){
			pageheader_container = '<div class="page-header-background" style="background-image: url(' + image_src + ') "/>'
			$pageheader.append(pageheader_container);
		}
	},

	initRightMenu: function(){
		if(!navbar_initialized){
			$navbar = $('nav').find('.navbar-collapse').first().clone(true);

			$sidebar = $('.sidebar');
			sidebar_color = $sidebar;//.data('color');

			$logo = $sidebar.find('.logo').first();
			logo_content = $logo[0].outerHTML;

			ul_content = '';

			//$navbar.attr('data-color',sidebar_color);

			//add the content from the regular header to the right
			$navbar.children('ul').each(function(){
				content_buff = $(this).html();
				ul_content = ul_content + content_buff;
			});

			//add the content from the sidebar to the right menu

			content_buff = $sidebar.find('.nav').html();
			ul_content = ul_content + content_buff;

			ul_content = '<div class="sidebar-wrapper">' +
							'<ul class="nav navbar-nav">' +
								ul_content +
							'</ul>'+
						'</div>';

			navbar_content = logo_content + ul_content;

			$navbar.html(navbar_content);

			$('body').append($navbar);

			/*background_image = $sidebar.data('image');
			 if(background_image != undefined){
                $navbar.css('background',"url('" + background_image + "')")
                       .removeAttr('data-nav-image')
                       .addClass('has-image');
            }*/


            $toggle = $('.navbar-toggler');

            $toggle.click(function(){
            	if(lbd.misc.navbar_menu_visible == 1){
            		$('html').removeClass('nav-open');
            		lbd.misc.navbar_menu_visible = 0;
            		$('#bodyClick').remove();
            		setTimeout(function(){
            			$toggle.removeClass('toggled');
            		}, 400);
            	} else {
            		setTimeout(function(){
            			$toggle.addClass('toggled');
            		},430);

            		div = '<div id="bodyClick"></div>';
            		$(div).appendTo("body").click(function(){
            			$('html').removeClass('nav-open');
            			lbd.misc.navbar_menu_visible = 0;
            			$('#bodyClick').remove();
            			setTimeout(function(){
            				$toggle.removeClass('toggled');
            			}, 400);
            		});

            		$('html').addClass('nav-open');
                    lbd.misc.navbar_menu_visible = 1;
            	}

            });
         	navbar_initialized = true;
		}
	}
}
 
 


gulpfile.js
// Gulp.js configuration
'use strict';

const productname = 'hermes-dashboard',
      version = 'v1.0.0';

const

  // source and build folders
  dir = {
    src         : './',
    build       : 'dist/'
  },

  // Gulp and plugins
  gulp          = require('gulp'),
  gutil         = require('gulp-util'),
  newer         = require('gulp-newer'),
  imagemin      = require('gulp-imagemin'),
  sass          = require('gulp-sass'),
  postcss       = require('gulp-postcss'),
  deporder      = require('gulp-deporder'),
  concat        = require('gulp-concat'),
  stripdebug    = require('gulp-strip-debug'),
  uglify        = require('gulp-uglify'),
  prettify      = require('gulp-jsbeautifier'),
  clean         = require('gulp-clean'),
  git           = require('gulp-git'), // https://www.npmjs.com/package/gulp-git
  zip           = require('gulp-zip');
;

// Browser-sync
var browsersync = false;

// Move HTML settings
const move_html = {
  src           : dir.src + '**/*.html',
  build         : dir.build
};

// copy HTML files
gulp.task('move_html', () => {
  return gulp.src(move_html.src)
    .pipe(newer(move_html.build))
    .pipe(gulp.dest(move_html.build));
});

// Move SASS folder
const move_sass = {
    src          : 'assets/_scss/**/*.scss',
    build        : dir.build + 'assets/sass/',
}

// copy Sass files
gulp.task('move_sass', () => {
  return gulp.src(move_sass.src)
    .pipe(newer(move_sass.build))
    .pipe(gulp.dest(move_sass.build));
});

// copy SASS parent
gulp.task('move_sass_parent', () => {
  return gulp.src('assets/css/*.scss')
    .pipe(newer(move_sass.build))
    .pipe(gulp.dest(move_sass.build));
});

// copy full assets
gulp.task('move_js', () => {
  return gulp.src('assets/js/**/*')
    .pipe(newer(dir.build + '/assets/js/'))
    .pipe(gulp.dest(dir.build + '/assets/js/'));
});

gulp.task('move_css', () => {
  return gulp.src('./assets/css/**/*')
    .pipe(newer(dir.build + '/assets/css/'))
    .pipe(gulp.dest(dir.build + '/assets/css/'));
});


gulp.task('move_fonts', () => {
  return gulp.src('./assets/fonts/**/*')
    .pipe(newer(dir.build + '/assets/fonts/'))
    .pipe(gulp.dest(dir.build + '/assets/fonts/'));
});


gulp.task('clean_scss', function () {
    return gulp.src(dir.build + '/assets/css/now-ui-dashboard.scss', {read: false})
        .pipe(clean());
});

// image settings
const images = {
  src         : dir.src + 'assets/img/**/*',
  build       : dir.build + 'assets/img/'
};

// image processing
gulp.task('images', () => {
  return gulp.src(images.src)
    .pipe(newer(images.build))
    .pipe(imagemin({ optimizationLevel: 8 }))
    .pipe(gulp.dest(images.build));
});

// CSS settings
var css = {
  src         : dir.src + 'scss/style.scss',
  watch       : dir.src + 'scss/**/*',
  build       : dir.build,
  sassOpts: {
    outputStyle     : 'nested',
    imagePath       : images.build,
    precision       : 3,
    errLogToConsole : true
  },
  processors: [
    require('postcss-assets')({
      loadPaths: ['images/'],
      basePath: dir.build,
      baseUrl: '/wp-content/themes/wptheme/'
    }),
    require('autoprefixer')({
      browsers: ['last 2 versions', '> 2%']
    }),
    require('css-mqpacker'),
    require('cssnano')
  ]
};

// CSS processing
gulp.task('css', ['images'], () => {
  return gulp.src(css.src)
    .pipe(sass(css.sassOpts))
    .pipe(postcss(css.processors))
    .pipe(gulp.dest(css.build))
    .pipe(browsersync ? browsersync.reload({ stream: true }) : gutil.noop());
});

// Zip files up
gulp.task('zip', function () {
 return gulp.src('dist/**/*')
  .pipe(zip(productname + '-' + version + '.zip'))
  .pipe(gulp.dest('.'));
});

// gulp.task('prettify', function() {
//   gulp.src('dist/**/*.html')
//     .pipe(prettify({indent_char: ' ', indent_size: 2}))
//     .pipe(gulp.dest('dist/'))
// });



// Move HTML settings
const live_demo = {
  src           : 'dist/**/*',
  build         : '../ct-freebies/public/' + productname + '/'
};

gulp.task('move_live_demo', () => {
  return gulp.src(live_demo.src)
    .pipe(newer(live_demo.build))
    .pipe(gulp.dest(live_demo.build));
});

gulp.task('prettify-html', function() {
  gulp.src(['./**/*.html'])
    .pipe(prettify({
        "html": {
            "allowed_file_extensions": ["htm", "html", "xhtml", "shtml", "xml", "svg"],
            "brace_style": "collapse", // [collapse|expand|end-expand|none] Put braces on the same line as control statements (default), or put braces on own line (Allman / ANSI style), or just put end braces on own line, or attempt to keep them where they are
            "end_with_newline": false, // End output with newline
            "indent_char": " ", // Indentation character
            "indent_handlebars": false, // e.g. {{#foo}}, {{/foo}}
            "indent_inner_html": false, // Indent <head> and <body> sections
            "indent_scripts": "normal", // [keep|separate|normal]
            "indent_size": 4, // Indentation size
            "max_preserve_newlines": 0, // Maximum number of line breaks to be preserved in one chunk (0 disables)
            "preserve_newlines": true, // Whether existing line breaks before elements should be preserved (only works before elements, not inside tags or for text)
            "unformatted": ["sub", "sup", "em", "strong", "i", "u", "strike", "big", "pre"], // List of tags that should not be reformatted
            "wrap_line_length": 0 // Lines should wrap at next opportunity after this number of characters (0 disables)
        }
    }))
    .pipe(gulp.dest('dist/'));
});

gulp.task('prettify-css', function() {
  gulp.src(['./assets/css/**/*.css'])
    .pipe(prettify({
        "css": {
            "allowed_file_extensions": ["css", "scss", "sass", "less"],
            "end_with_newline": false, // End output with newline
            "indent_char": " ", // Indentation character
            "indent_size": 4, // Indentation size
            "newline_between_rules": true, // Add a new line after every css rule
            "selector_separator": " ",
            "selector_separator_newline": true // Separate selectors with newline or not (e.g. "a,\nbr" or "a, br")
        }
    }))
    .pipe(gulp.dest('dist/assets/css/'));
});

gulp.task('prettify-js', function() {
  gulp.src(['./assets/js/**/*.js'])
    .pipe(prettify({
        "js": {
            "allowed_file_extensions": ["js", "json", "jshintrc", "jsbeautifyrc"],
            "brace_style": "collapse", // [collapse|expand|end-expand|none] Put braces on the same line as control statements (default), or put braces on own line (Allman / ANSI style), or just put end braces on own line, or attempt to keep them where they are
            "break_chained_methods": false, // Break chained method calls across subsequent lines
            "e4x": false, // Pass E4X xml literals through untouched
            "end_with_newline": false, // End output with newline
            "indent_char": " ", // Indentation character
            "indent_level": 0, // Initial indentation level
            "indent_size": 4, // Indentation size
            "indent_with_tabs": false, // Indent with tabs, overrides `indent_size` and `indent_char`
            "jslint_happy": false, // If true, then jslint-stricter mode is enforced
            "keep_array_indentation": false, // Preserve array indentation
            "keep_function_indentation": false, // Preserve function indentation
            "max_preserve_newlines": 0, // Maximum number of line breaks to be preserved in one chunk (0 disables)
            "preserve_newlines": true, // Whether existing line breaks should be preserved
            "space_after_anon_function": false, // Should the space before an anonymous function's parens be added, "function()" vs "function ()"
            "space_before_conditional": true, // Should the space before conditional statement be added, "if(true)" vs "if (true)"
            "space_in_empty_paren": false, // Add padding spaces within empty paren, "f()" vs "f( )"
            "space_in_paren": false, // Add padding spaces within paren, ie. f( a, b )
            "unescape_strings": false, // Should printable characters in strings encoded in \xNN notation be unescaped, "example" vs "\x65\x78\x61\x6d\x70\x6c\x65"
            "wrap_line_length": 0 // Lines should wrap at next opportunity after this number of characters (0 disables)
        }
    }))
    .pipe(gulp.dest('dist/assets/js'));
});


gulp.task('prettify',['prettify-html','prettify-css','prettify-js'], () => {
    gutil.log('Finished prettify');
});


// run all tasks
gulp.task('build', ['prettify','move_html', 'move_css','move_js','move_sass_parent','move_sass', 'images', 'move_fonts']);

// run all tasks
// gulp.task('live_demo', ['move_html', 'move_css','move_js','move_sass_parent','move_sass', 'images', 'move_fonts','clean_scss']);


assets/js/hd.js

var navbar_initialized = false;

var seq = 0, delays = 80, durations = 500;
(function(){

       // if we are on windows OS we activate the perfectScrollbar function
       $(' .sidebar-wrapper, .sidebar-wrapper, .main-panel').perfectScrollbar();

       $('html').addClass('perfect-scrollbar-on');
  	
  		if($("html").hasClass("nav-open")){
  			$('.collapse').perfectScrollbar();
  		}
})();

$(document).ready(function(){
	window_width = $(window).width();

	hd.checkHeaderImage();
	
	if(window_width >= 992){
		$(document).on('ps-scroll-x', function () {
	  		hermesdashboard.checkScrollForParallax();
		});
	}
	
	if(window_width <= 991){
		hd.initRightMenu();
	}

	$(document).on('ps-scroll-x', function () {
  		hermesdashboard.checkScrollForParallax();
	});
	var scroll_start = 0;
	var startchange=$('.wave');
	var offset = startchange.offset();
	$(document).scroll(function(){
		scroll_start = $(this).scrollTop();
		if(scroll_start > offset.top){
			$(".navbar").css('background-color', 'black');
		} else {
			$(".navbar").css('background-color', 'transparent');
		}
	});

	$('a[data-toggle=tab').on('shown.bs.tab', function (e) {
 	 window.dispatchEvent(new Event('resize'));   
	});
});

$(window).resize(function(){
	if($(window).width() <= 991){
		hd.initRightMenu();
	}
})



hd = {

	misc:{
		navbar_menu_visible: 0 
	},

	checkHeaderImage: function(){
		$pageheader = $('.page-header');
		image_src = $pageheader.data('image');

		if(image_src !== undefined){
			pageheader_container = '<div class="page-header-background" style="background-image: url(' + image_src + ') "/>'
			$pageheader.append(pageheader_container);
		}
	},

	initRightMenu: function(){
		if(!navbar_initialized){
			$navbar = $('nav').find('.navbar-collapse').first().clone(true);

			$sidebar = $('.sidebar');

			$logo = $sidebar.find('.logo').first();
			logo_content = $logo[0].outerHTML;

			ul_content = '';


			//add the content from the regular header to the right
			$navbar.children('ul').each(function(){
				content_buff = $(this).html();
				ul_content = ul_content + content_buff;
			});

			//add the content from the sidebar to the right menu

			content_buff = $sidebar.find('.nav').html();
			ul_content = ul_content + content_buff;

			ul_content = '<div class="sidebar-wrapper">' +
							'<ul class="nav navbar-nav">' +
								ul_content +
							'</ul>'+
						'</div>';

			navbar_content = logo_content + ul_content;

			$navbar.html(navbar_content);

			$('body').append($navbar);

            $toggle = $('.navbar-toggler');

            $toggle.click(function(){
            	if(hd.misc.navbar_menu_visible == 1){
            		$('html').removeClass('nav-open');
            		hd.misc.navbar_menu_visible = 0;
            		$('#bodyClick').remove();
            		setTimeout(function(){
            			$toggle.removeClass('toggled');
            		}, 400);
            	} else {
            		setTimeout(function(){
            			$toggle.addClass('toggled');
            		},430);

            		div = '<div id="bodyClick"></div>';
            		$(div).appendTo("body").click(function(){
            			$('html').removeClass('nav-open');
            			hd.misc.navbar_menu_visible = 0;
            			$('#bodyClick').remove();
            			setTimeout(function(){
            				$toggle.removeClass('toggled');
            			}, 400);
            		});

            		$('html').addClass('nav-open');
                    hd.misc.navbar_menu_visible = 1;
            	}

            });
         	navbar_initialized = true;
		}
	},

	startAnimationForLineChart: function(chart){

        chart.on('draw', function(data) {
          if(data.type === 'line' || data.type === 'area') {
            data.element.animate({
              d: {
                begin: 600,
                dur: 700,
                from: data.path.clone().scale(1, 0).translate(0, data.chartRect.height()).stringify(),
                to: data.path.clone().stringify(),
                easing: Chartist.Svg.Easing.easeOutQuint
              }
            });
          } else if(data.type === 'ct-circle') {
                seq++;
                data.element.animate({
                  opacity: {
                    begin: seq * delays,
                    dur: durations,
                    from: 0,
                    to: 1,
                    easing: 'ease'
                  }
                });
            }
        });

        seq = 0;
    },
    //replace the point with a circle so you can add outline
    chartistPointWithMargin: function(chart){
	    	chart.on('draw', function(data) {
		      if(data.type === 'point') {
		        var circle = new Chartist.Svg('circle', {
		          cx: [data.x], cy:[data.y], r:[7],
		        }, 'ct-circle');
		        data.element.replace(circle);
	      }
	    });
    }


}
 
var big_image;
hermesdashboard = {
	checkScrollForParallax: debounce(function(){
		var curent_scroll = $('.main-panel').scrollTop();
		if($("html").hasClass("perfect-scrollbar-on")){
			oVal = ($('.main-panel').scrollTop() / 3);
		} else {
			oVal = ($(window).scrollTop() / 3);	
		}
		big_image.css({
            'transform':'translate3d(0,' + oVal +'px,0)',
            '-webkit-transform':'translate3d(0,' + oVal +'px,0)',
            '-ms-transform':'translate3d(0,' + oVal +'px,0)',
            '-o-transform':'translate3d(0,' + oVal +'px,0)'
        });
	}, 6)
};



function debounce(func, wait, immediate) {
	var timeout;
	return function() {
		var context = this, args = arguments;
		clearTimeout(timeout);
		timeout = setTimeout(function() {
			timeout = null;
			if (!immediate) func.apply(context, args);
		}, wait);
		if (immediate && !timeout) func.apply(context, args);
	};
};

js/jquery.sharrre.js
/*!
 *  Sharrre.com - Make your sharing widget!
 *  Version: 2.0.1
 *  Author: Julien Hany
 *  License: MIT http://en.wikipedia.org/wiki/MIT_License or GPLv2 http://en.wikipedia.org/wiki/GNU_General_Public_License
 */

(function ($, window, document, undefined) {

    /* Defaults*/
    var pluginName = 'sharrre',
        defaults = {
            className: 'sharrre',
            share: {},
            shareTotal: 0,
            template: '',
            title: '',
            url: document.location.href,
            text: document.title,
            urlCurl: 'sharrre.php',  //PHP script for google plus...
            count: {}, //counter by social network
            total: 0,  //total of sharing
            shorterTotal: true, //show total by k or M when number is to big
            enableHover: true, //disable if you want to personalize hover event with callback
            enableCounter: true, //disable if you just want use buttons
            enableTracking: false, //tracking with google analitycs
            defaultUrl: "javascript:void(0);",
            popup: { // Set the popup width and height
                width: 900,
                height: 500
            },
            hover: function () {
            }, //personalize hover event with this callback function
            hide: function () {
            }, //personalize hide event with this callback function
            click: function () {
            }, //personalize click event with this callback function
            render: function () {
            }
        };

    /* Plugin constructor*/
    function Plugin(element, options) {
        this.element = element;
        this.options = $.extend(true, {}, defaults, options);
        this.options.share = options.share; //simple solution to allow order of buttons
        this._defaults = defaults;
        this._name = pluginName;
        this.platforms = {};
        this.init();
    };

    /* Initialization method
     ================================================== */
    Plugin.prototype.init = function () {
        var self = this;

        // Load enabled platforms
        $.each(self.options.share, function (name, val) {
            if (val === true) {
                self.platforms[name] = SharrrePlatform.get(name, self.options.buttons[name]);
            }
        });

        $(this.element).addClass(this.options.className); //add class

        //HTML5 Custom data
        if (typeof $(this.element).data('title') !== 'undefined') {
            this.options.title = $(this.element).attr('data-title');
        }
        if (typeof $(this.element).data('url') !== 'undefined') {
            this.options.url = $(this.element).data('url');
        }
        if (typeof $(this.element).data('text') !== 'undefined') {
            this.options.text = $(this.element).data('text');
        }

        //how many social website have been selected
        $.each(this.options.share, function (name, val) {
            if (val === true) {
                self.options.shareTotal++;
            }
        });

        if (self.options.enableCounter === true) {  //if for some reason you don't need counter
            //get count of social share that have been selected
            $.each(this.options.share, function (name, val) {
                if (val === true) {
                    //self.getSocialJson(name);
                    try {
                        self.getSocialJson(name);
                    } catch (e) {
                    }
                }
            });
        } else if (self.options.template !== '') {
            self.renderer();
            self.options.count[name] = 0;
            self.rendererPerso();
        }

        if (self.options.template !== '') {  //for personalized button (with template)
            this.options.render(this, this.options);
        }
        else { // if you want to use official button like example 3 or 5
            this.loadButtons();
        }

        //add hover event
        $(this.element).on('mouseenter', function () {
            //load social button if enable and 1 time
            if ($(this).find('.buttons').length === 0 && self.options.enableHover === true) {
                self.loadButtons();
            }
            self.options.hover(self, self.options);
        }).on('mouseleave', function () {
            self.options.hide(self, self.options);
        });

        //click event
        $(this.element).click(function (e) {
            e.preventDefault();
            self.options.click(self, self.options);
            return false;
        });
    };

    /* loadButtons methode
     ================================================== */
    Plugin.prototype.loadButtons = function () {
        var self = this;
        $(this.element).append('<div class="buttons"></div>');
        $.each(self.options.share, function (name, val) {

            if (val == true) {
                self.platforms[name].load(self);
                if (self.options.enableTracking === true) { //add tracking
                    self.platforms[name].tracking();
                }
            }
        });
    };

    /* getSocialJson methode
     ================================================== */
    Plugin.prototype.getSocialJson = function (name) {
        var self = this,
            count = 0,
            settings = self.platforms[name].settings,
            buttonUrl = self.platforms[name].url(this.options.urlCurl),
            replaceUrl = encodeURIComponent(this.options.url);
        if (settings.url.length) {
            buttonUrl = settings.url;
        }
        if (settings.urlCount === true && buttonUrl !== '') {
            replaceUrl = buttonUrl;
        }
        if (settings.count === false) {
            buttonUrl = '';
        }
        url = buttonUrl.replace('{url}', replaceUrl);
        if (url != '') {  //urlCurl = '' if you don't want to used PHP script but used social button
            $.getJSON(url, function (json) {
                if (typeof json.count !== "undefined") {  //GooglePlus, Stumbleupon, Twitter, Pinterest and Digg
                    var temp = json.count + '';
                    temp = temp.replace('\u00c2\u00a0', '');  //remove google plus special chars
                    count += parseInt(temp, 10);
                }
                //get the FB total count (shares, likes and more)
                else if (json.data && json.data.length > 0 && typeof json.data[0].total_count !== "undefined") { //Facebook total count
                    count += parseInt(json.data[0].total_count, 10);
                }
                else if (typeof json[0] !== "undefined") {  //Delicious
                    count += parseInt(json[0].total_posts, 10);
                }
                else if (typeof json[0] !== "undefined") {  //Stumbleupon
                }
                self.options.count[name] = count;
                self.options.total += count;
                self.renderer();
                self.rendererPerso();
            })
                .error(function () {
                    self.options.count[name] = 0;
                    self.rendererPerso();
                });
        }
        else {
            self.renderer();
            self.options.count[name] = 0;
            self.rendererPerso();
        }
    };

    /* launch render methode
     ================================================== */
    Plugin.prototype.rendererPerso = function () {
        //check if this is the last social website to launch render
        var shareCount = 0;
        for (e in this.options.count) {
            shareCount++;
        }
        if (shareCount === this.options.shareTotal) {
            this.options.render(this, this.options);
        }
    };

    /* render methode
     ================================================== */
    Plugin.prototype.renderer = function () {
        var total = this.options.total,
            template = this.options.template;
        if (this.options.shorterTotal === true) {  //format number like 1.2k or 5M
            total = this.shorterTotal(total);
        }

        if (template !== '') {  //if there is a template
            template = template.replace('{total}', total);
            $(this.element).html(template);
        }
        else { //template by defaults
            $(this.element).html(
                '<div class="box"><a class="count" href="' + this.options.defaultUrl + '">' + total + '</a>' +
                (this.options.title !== '' ? '<a class="share" href="' + this.options.defaultUrl + '">' + this.options.title + '</a>' : '') +
                '</div>'
            );
        }
    };

    /* format total numbers like 1.2k or 5M
     ================================================== */
    Plugin.prototype.shorterTotal = function (num) {
        if (num >= 1e6) {
            num = (num / 1e6).toFixed(2) + "M"
        } else if (num >= 1e3) {
            num = (num / 1e3).toFixed(1) + "k"
        }
        return num;
    };

    /* Methode for open popup
     ================================================== */
    Plugin.prototype.openPopup = function (site) {
        this.platforms[site].popup(this.options);  //open
        if (this.options.enableTracking === true) { //tracking!
            infos = this.platforms[site].trackingAction;
            _gaq.push(['_trackSocial', infos.site, infos.action]);
        }
    };

    /* Methode for add +1 to a counter
     ================================================== */
    Plugin.prototype.simulateClick = function () {
        var html = $(this.element).html();
        $(this.element).html(html.replace(this.options.total, this.options.total + 1));
    };

    /* Methode for add +1 to a counter
     ================================================== */
    Plugin.prototype.update = function (url, text) {
        if (url !== '') {
            this.options.url = url;
        }
        if (text !== '') {
            this.options.text = text;
        }
    };

    /* A really lightweight plugin wrapper around the constructor, preventing against multiple instantiations
     ================================================== */
    $.fn[pluginName] = function (options) {
        var args = arguments;
        if (options === undefined || typeof options === 'object') {
            return this.each(function () {
                if (!$(this).data('plugin_' + pluginName)) {
                    $(this).data('plugin_' + pluginName, new Plugin(this, options));
                }
            });
        } else if (typeof options === 'string' && options[0] !== '_' && options !== 'init') {
            return this.each(function () {
                var instance = $(this).data('plugin_' + pluginName);
                if (instance instanceof Plugin && typeof instance[options] === 'function') {
                    instance[options].apply(instance, Array.prototype.slice.call(args, 1));
                }
            });
        }
    };
})(window.jQuery || window.Zepto, window, document);


assets/js/demo.js
type = ['','info','success','warning','danger'];
    	
var seq = 0, delays = 80, durations = 500;
demo = {
   initPickColor: function(){
        $('.pick-class-label').click(function(){
            var new_class = $(this).attr('new-class');  
            var old_class = $('#display-buttons').attr('data-class');
            var display_div = $('#display-buttons');
            if(display_div.length) {
            var display_buttons = display_div.find('.btn');
            display_buttons.removeClass(old_class);
            display_buttons.addClass(new_class);
            display_div.attr('data-class', new_class);
            }
        });
    },
    
    initChartist: function(){    
        
        var data_year =  {
         labels: ['JAN.', 'FEB.', 'MAR.', 'APR.', 'MAY.', 'JUN.', 'JUL.', 'AUG.', 'SEP.', 'OCT.', 'NOV.', 'DEC.'],
          series: [
            [0, 180, 80, 320, 220, 420, 190, 570, 440, 370, 420, 260]
          ]
        };
        var options = {
            lineSmooth: Chartist.Interpolation.simple({
                    divisor: 100
                  }),
                  showArea: false,
                  showPoint: true,
                  showLine: true,
                  height: '200px',
                  axisX: {
                    showGrid: false
                  }
        };

      var chart_year = new Chartist.Line('#chartPreferences1', data_year, options);
      hd.chartistPointWithMargin(chart_year);
      hd.startAnimationForLineChart(chart_year);
      


      var data_month = {
         labels: ['1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20','21','22','23','24','25','26','27','28','29','30'],
          series: [
            [12, 16, 21, 3, 12, 0, 22, 10, 6, 18, 9, 13, 36, 4, 12, 17, 19, 10, 10, 1, 8, 12, 2, 22, 50, 42, 2, 18, 10, 5]
          ]
        };
     

        var chart_month = new Chartist.Line('#chartPreferences2', data_month, options);

        hd.startAnimationForLineChart(chart_month);
        hd.chartistPointWithMargin(chart_month);

        var data_day = {
         labels: ['1', '2', '3', '4', '5', '6', '7'],
          series: [
            [12, 3, 22, 16, 17, 2, 4]
          ]
        };

      var chart_day = new Chartist.Line('#chartPreferences3', data_day, options);
      hd.startAnimationForLineChart(chart_day);
      hd.chartistPointWithMargin(chart_day);

      var data_charges =  {
        series: [
            [7,9,7.3,7.7,8.6,10,9.1,8.9,9,8.6,7.3,7.8,8.2]
          ]
        };
      var options2 = {
        lineSmooth: Chartist.Interpolation.simple({
                divisor: 100
              }),
              showArea: false,
              showPoint: true,
              low: 6,
              high:11,
              showLine: true,
              height: '150px',
              axisX: {
                showGrid: false
              }
    };
      

    var chart_charges = new Chartist.Line('#chartCharges', data_charges, options2);
    hd.chartistPointWithMargin(chart_charges);

    var data_customers =  {
    series: [
        [8.9, 9.5, 10,9.2, 8.9, 9, 8.7, 7.2, 8.8,9.8, 8.9,10]
      ]
    };     
    
    var chart_customers = new Chartist.Line('#chartCustomers', data_customers, options2);
    hd.chartistPointWithMargin(chart_customers);

    },

    presentationChartist: function(){

       var data_presentation =  {
    series: [
        [2, 5, 4.5, 6, 5, 7, 6, 7, 5.5, 6, 5, 4, 2]
      ]
    };

    var options_presentation = {
        lineSmooth: Chartist.Interpolation.simple({
                divisor: 2
              }),
              low: 0,
              showArea: true,
              showPoint: true,
              showLine: true,
              height: "400px",
              width: "2000px",
              axisX: {
                showGrid: false
              },
              axisY: {
                showGrid: false
              }
    };
      
      var chart_presentation = new Chartist.Line('#chartPresentation',data_presentation, options_presentation);
      hd.startAnimationForLineChart(chart_presentation);
      hd.chartistPointWithMargin(chart_presentation);
      
    },
    
    initGoogleMaps: function(){
        var myLatlng = new google.maps.LatLng(40.748817, -73.985428);
        var mapOptions = {
          zoom: 13,
          center: myLatlng,
          scrollwheel: false, //we disable de scroll over the map, it is a really annoing when you scroll through page
          styles: [{"featureType":"water","stylers":[{"saturation":43},{"lightness":-11},{"hue":"#0088ff"}]},{"featureType":"road","elementType":"geometry.fill","stylers":[{"hue":"#ff0000"},{"saturation":-100},{"lightness":99}]},{"featureType":"road","elementType":"geometry.stroke","stylers":[{"color":"#808080"},{"lightness":54}]},{"featureType":"landscape.man_made","elementType":"geometry.fill","stylers":[{"color":"#ece2d9"}]},{"featureType":"poi.park","elementType":"geometry.fill","stylers":[{"color":"#ccdca1"}]},{"featureType":"road","elementType":"labels.text.fill","stylers":[{"color":"#767676"}]},{"featureType":"road","elementType":"labels.text.stroke","stylers":[{"color":"#ffffff"}]},{"featureType":"poi","stylers":[{"visibility":"off"}]},{"featureType":"landscape.natural","elementType":"geometry.fill","stylers":[{"visibility":"on"},{"color":"#b8cb93"}]},{"featureType":"poi.park","stylers":[{"visibility":"on"}]},{"featureType":"poi.sports_complex","stylers":[{"visibility":"on"}]},{"featureType":"poi.medical","stylers":[{"visibility":"on"}]},{"featureType":"poi.business","stylers":[{"visibility":"simplified"}]}]
    
        }
        var map = new google.maps.Map(document.getElementById("map"), mapOptions);
        
        var marker = new google.maps.Marker({
            position: myLatlng,
            title:"Hello World!"
        });
        
        // To add the marker to the map, call setMap();
        marker.setMap(map);
    },
    
	showNotification: function(from, align){
    	color = Math.floor((Math.random() * 4) + 1);
    	
    	$.notify({
        	icon: "nc-icon nc-bell-55",
        	message: "Welcome to <b>Hermes Dashboard</b> - a beautiful freebie for every web developer."
        	
        },{
            type: type[color],
            timer: 400,
            placement: {
                from: from,
                align: align
            }
        });
	}

    
}


